tests/meta/upload/scout/test_meta_upload_scoutapi_rna.py

```diff
@@ -254,3 +254,173 @@ def test_upload_splice_junctions_bed_to_scout_no_subject_id(
+    upload_scout_api: UploadScoutAPI,
+):
+    """Test that A RNA case's gene fusion report and junction splice files for all samples is not uploaded if the is_tumour is not matching"""
+
+    # GIVEN a sample in the RNA case is NOT connected to a sample in the DNA case via is_tumour (i.e. different is_tumour)
+    for link in rna_store.family(rna_case_id).links:
+        link.sample.is_tumour = True
+    for link in rna_store.family(dna_case_id).links:
+        link.sample.is_tumour = False
+    rna_store.commit()
+    upload_scout_api.status_db = rna_store
+
+    # GIVEN the connected RNA sample has a bigWig in Housekeeper
+
+    # WHEN running the method to upload RNA files to Scout
+    caplog.set_level(logging.INFO)
+    # THEN an exception should be raised on unconnected data
+    with pytest.raises(CgDataError):
+        upload_scout_api.upload_rna_coverage_bigwig_to_scout(case_id=rna_case_id, dry_run=True)
+
+
+def test_upload_splice_junctions_bed_to_scout_tumour_non_matching(
+    caplog: Generator[LogCaptureFixture, None, None],
+    dna_case_id: str,
+    mip_rna_analysis_hk_api: HousekeeperAPI,
+    rna_case_id: str,
+    rna_store: Store,
+    upload_scout_api: UploadScoutAPI,
+):
+    """Test that A RNA case's junction splice files for all samples is not uploaded if the is_tumour is not matching"""
+
+    # GIVEN a sample in the RNA case is NOT connected to a sample in the DNA case via is_tumour (i.e. different is_tumour)
+    for link in rna_store.family(rna_case_id).links:
+        link.sample.is_tumour = True
+    for link in rna_store.family(dna_case_id).links:
+        link.sample.is_tumour = False
+    rna_store.commit()
+    upload_scout_api.status_db = rna_store
+
+    # GIVEN the connected RNA sample has a junction bed in Housekeeper
+
+    # WHEN running the method to upload RNA files to Scout
+    caplog.set_level(logging.INFO)
+    # THEN an exception should be raised on unconnected data
+    with pytest.raises(CgDataError):
+        upload_scout_api.upload_splice_junctions_bed_to_scout(case_id=rna_case_id, dry_run=True)
+
+
+def test_upload_rna_fusion_report_to_scout_tumour_multiple_matches(
+    caplog: Generator[LogCaptureFixture, None, None],
+    dna_case_id: str,
+    helpers: StoreHelpers,
+    mip_rna_analysis_hk_api: HousekeeperAPI,
+    rna_case_id: str,
+    rna_store: Store,
+    upload_scout_api: UploadScoutAPI,
+):
+    """Test that an RNA case's gene fusion report is not uploaded if the is_tumour has too many DNA-matches"""
+
+    # GIVEN a sample in the RNA case is connected to a sample in the DNA case via is_tumour (i.e. same is_tumour)
+    ensure_two_dna_tumour_matches(dna_case_id, helpers, rna_case_id, rna_store)
+    upload_scout_api.status_db = rna_store
+
+    # GIVEN the connected RNA case has a research fusion report in Housekeeper
+
+    # WHEN running the method to upload RNA files to Scout
+    caplog.set_level(logging.INFO)
+    # THEN an exception should be raised on unconnected data
```

Comment:
```suggestion

    # THEN an exception should be raised on unconnected data
```
Is the same error raised as unconnected? Should it not be multiple tumours errors or similar? Or maybe the log will tell you this